### PR TITLE
CMSIS-NN : MVE support for s8 depthwise conv

### PR DESCRIPTION
**Pull request by felix-johnny**

### CODE REVIEW

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_s8_opt.c [hunk 1]
```diff
@@ -75,20 +75,114 @@ arm_status arm_depthwise_conv_s8_opt(const q7_t *input,
...
     /* Check input constraints input_ch == output_ch */
     if (input_ch != output_ch)
     {
         return ARM_MATH_SIZE_MISMATCH;
     }
+#ifdef ARM_MATH_MVEI
+    (void)dilation_x;
+    (void)dilation_y;
+
+    /* Generate two columns from the input tensor a GEMM computation */
```
Comment (freddan80) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_s8_opt.c]: word missing?

File: CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_s8_opt.c [hunk 2]
```diff
@@ -306,7 +399,7 @@ arm_status arm_depthwise_conv_s8_opt(const q7_t *input,
                                  dilation_x,
                                  dilation_y,
                                  NULL);
-#endif /* ARM_MATH_DSP & ARM_MATH_LOOPUNROLL*/
+#endif /* ARM_MATH_MVEI */
```
Comment (freddan80) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_s8_opt.c]: | (ARM_MATH_DSP & ARM_MATH_LOOPUNROLL)

File: CMSIS/NN/Include/arm_nnsupportfunctions.h [hunk 3]
```diff
@@ -167,6 +167,37 @@ void arm_q7_to_q15_reordered_with_offset(const q7_t *src, q15_t *dst, uint32_t b
...
+ * @param[in]    col     pointer to im2col buffer, always consists of 2 columns.
+ * @param[in]    num_ch   number of channels
+ * @param[in]    out_shift  pointer to per output channel requantization shift parameter.
+ * @param[in]    out_mult   pointer to per output channel requantization multiplier parameter.
+ * @param[in]    out_offset      output tensor offset.
+ * @param[in]    activation_min   minimum value to clamp the output to. Range : int8
+ * @param[in]    activation_max   maximum value to clamp the output to. Range : int8
+ * @param[in]    kernel_size   number of elements in one column.
+ * @param[in]    output_bias per output channel bias. Range : int32
+ * @param[in,out]   out_0       pointer to output
```
Comment (freddan80) [inline CMSIS/NN/Include/arm_nnsupportfunctions.h]: out_0? perhaps clarify by splitting up. So:
@param out [in] is the output pointer before operation 
@return: leave as it is now

File: CMSIS/NN/Source/ConvolutionFunctions/arm_nn_depthwise_conv_s8_core.c [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+        col_1_sum = vaddq_n_s32(col_1_sum, out_offset);
+        col_1_sum = vmaxq_s32(col_1_sum, vdupq_n_s32(activation_min));
+        col_1_sum = vminq_s32(col_1_sum, vdupq_n_s32(activation_max));
+        vstrbq_p_s32(out_tmp + num_ch, col_1_sum, p);
+
+        out_tmp += tail_ch;
+    }
+
+    return out_tmp + num_ch;
```
Comment (freddan80) [inline CMSIS/NN/Source/ConvolutionFunctions/arm_nn_depthwise_conv_s8_core.c]: why add num_ch?

File: CMSIS/NN/Include/arm_nnsupportfunctions.h (hunk 3 above)
Comment (felix-johnny) [reply CMSIS/NN/Include/arm_nnsupportfunctions.h]: @freddan80 Re-checked on the Doxygen defintion for in, out, in/out. This should be just an 'out'.

File: CMSIS/NN/Source/ConvolutionFunctions/arm_nn_depthwise_conv_s8_core.c (hunk 4 above)
Comment (felix-johnny) [reply CMSIS/NN/Source/ConvolutionFunctions/arm_nn_depthwise_conv_s8_core.c]: We process two columns at at time and each column implies num_ch number of elements..This is to increment the out pointer over the second num_ch number of elements.

**Pull request merged**